### PR TITLE
introduce base class for EKS sensors

### PR DESCRIPTION
**Pull request by vandonr-amz**
most of the EKS sensors do the same thing, this can be grouped in one base class, a bit like what was done here
https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/sensors/sagemaker.py#L33

### CODE REVIEW

**Comment by o-nikolas** [comment]
@ferruzzi as our resident EKS expert, do you have time to take a look?

File: airflow/providers/amazon/aws/sensors/eks.py [hunk 1]
```diff
@@ -57,7 +58,71 @@
...
+        maintained on each worker node).
+    :param region: Which AWS region the connection should use.
+        If this is None or empty then the default boto3 behaviour is used.
+    """
+
+    def __init__(
+        self,
+        *,
+        cluster_name: str,
+        target_state: Any,
```
Comment (ferruzzi) [inline airflow/providers/amazon/aws/sensors/eks.py]: I think I would prefer to see this as 
```suggestion
        target_state: ClusterStates | NodegroupStates | FargateProfileStates,
```

and modify the return type hinting on the get_state() methods below, but I can also see how that would be getting a bit too verbose, so I'll leave it up to you.  Maybe a new Type:

```
from typing import NewType
TargetStates = NewType('TargetStates', ClusterStates | NodegroupStates | FargateProfileStates)
```

here or in the hook?  Just a thought (non-blocking)

**Review by ferruzzi — APPROVED** [review]
Left a non-blocking thought.  Otherwise I like it.  I love seeing commits that reduce linecount!

File: airflow/providers/amazon/aws/sensors/eks.py (hunk 1 above)
Comment (vandonr-amz) [reply airflow/providers/amazon/aws/sensors/eks.py]: I would have liked to write a class with generics here, but python doesn't allow that :(

Comment (ferruzzi) [reply airflow/providers/amazon/aws/sensors/eks.py]: In python you can do it like this:

```
from typing import TypeVar

T = TypeVar('T')      # Declare type variable

def first(l: Sequence[T]) -> T:   # Generic function
    return l[0]
```

But at that point I honestly don't know how that is better than just using 'Any'.

Comment (vandonr-amz) [reply airflow/providers/amazon/aws/sensors/eks.py]: the problem is that I think you can templatize a function but not a class.

**Review by o-nikolas — APPROVED** [review]

**Pull request merged**